python/lsst/ap/pipe/matchSourceInjected.py

```diff
@@ -0,0 +1,268 @@
+# This file is part of ap_pipe.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["MatchInitialPVIInjectedTask",
+           "MatchInitialPVIInjectedConfig"]
+
+import astropy.units as u
+import numpy as np
+from scipy.spatial import cKDTree
+
+from lsst.geom import Box2D
+import lsst.pex.config as pexConfig
+from lsst.pipe.base import PipelineTask, PipelineTaskConnections, Struct
+import lsst.pipe.base.connectionTypes as connTypes
+
+from lsst.source.injection import VisitInjectConfig
+
+
+class MatchInitialPVIInjectedConnections(
+        PipelineTaskConnections,
+        defaultTemplates={"coaddName": "deep",
+                          "fakesType": "fakes_",
+                          "injection_prefix": "injection_",
+                          "injected_prefix": "injected_"},
+        dimensions=("instrument",
+                    "visit",
+                    "detector")):
+    injectedInitialPVICat = connTypes.Input(
+        doc="Catalog of sources injected in the images.",
+        name="{fakesType}initial_pvi_catalog",
+        storageClass="ArrowAstropy",
+        dimensions=("instrument", "visit", "detector"),
+        deferLoad=False,
+        multiple=False
+    )
+    diffIm = connTypes.Input(
+        doc="Difference image on which the DiaSources were detected.",
+        name="{fakesType}{coaddName}Diff_differenceExp",
+        storageClass="ExposureF",
+        dimensions=("instrument", "visit", "detector"),
+    )
+    associatedDiaSources = connTypes.Input(
+        doc="A DiaSource catalog to match against fakeCat. Assumed "
+            "to be SDMified.",
+        name="{fakesType}{coaddName}Diff_assocDiaSrc",
+        storageClass="DataFrame",
+        dimensions=("instrument", "visit", "detector"),
+    )
+    matchedDiaSources = connTypes.Output(
+        doc="A catalog of those fakeCat sources that have a match in "
+            "associatedDiaSources. The schema is the union of the schemas for "
+            "``fakeCat`` and ``associatedDiaSources``.",
+        name="{fakesType}{coaddName}Diff_matchDiaSrc",
+        storageClass="DataFrame",
```

Comment:
Could this be `ArrowAstropy` instead? We are trying to move away from DataFrames.

Reply:
I tried but we have found a bug on serializing the datetime format in parquet with PyArrow. Will file a new ticket to make this change once this bug is solved.